File: LAB_3/LAB_3_Java/src/Producer.java
public class Producer implements Runnable
{
    private final int _maxItem;
    private final Storage _storage;

    public Producer(int maxItem, Storage storage)
    {
        _maxItem = maxItem;
        _storage = storage;
    }

    @Override
    public void run()
    {
        for(int i = 0; i< _maxItem; i++)
        {
            try
            {
                _storage.acquireFull();
                _storage.acquireAccess();

                _storage.addItem("item " + i);
                System.out.println("Added item " + i);

                _storage.releaseAccess();
                _storage.releaseEmpty();
            }
            catch (InterruptedException e)
            {
                e.printStackTrace();
            }
        }
    }
}

File: LAB_3/LAB_3_Csharp/LAB_3_Csharp/Producer.cs
using System;

namespace LAB_3_Csharp
{
    internal class Producer
    {
        private int _maxItem = 0;
        private Storage _storage;
        public Producer(int maxItem, Storage storage)
        {
            _maxItem = maxItem;
            _storage = storage;
        }

        public void PutItem()
        {
            for (int i = 0; i < _maxItem; i++)
            {
                _storage.AcquireFull();
                _storage.AcquireAccess();

                _storage.AddItem("item " + i);
                Console.WriteLine("Added item " + i);

                _storage.ReleaseAccess();
                _storage.ReleaseEmpty();

            }
        }
    }
}
